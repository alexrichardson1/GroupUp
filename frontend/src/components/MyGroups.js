import NavBar from "components/NavBar";
import axios from "axios";
import { config } from "Constants";
import { UserContext } from "components/auth/UserContext";
import { Component } from "react";
import { LinkContainer } from "react-router-bootstrap";
import { Button, Card, ListGroup } from "react-bootstrap";

export default class MyGroups extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filteredGroups: [],
    };
  }

  static contextType = UserContext;

  async getGroups() {
    let result = "";
    await axios
      .get(`${config.API_URL}/group/`)
      .then((res) => {
        const group = res.data;
        result = group;
      })
      .catch((error) => {
        console.error(error);
      });

    return result;
  }

  async componentDidMount() {
    this.setState({ filteredGroups: await this.getGroups() }, () =>
      this.filterMyGroups()
    );
  }

  filterMyGroups = () => {
    let value = this.context;
    const user = value.value;
    console.log(this.state.filteredGroups);
    console.log("groups ^");
    this.setState({
      filteredGroups: this.state.filteredGroups.filter(
        (gr) => gr.leader === user || gr.teammates.includes(user.split(" ")[0])
      ),
    });
  };

  render() {
    return (
      <div>
        <NavBar
          renderBool={[false, false, false, false]}
          create={false}
          savedSearches={true}
        />
        <h3 className="groupsHome">Groups you're already in.</h3>
        {this.state.filteredGroups.map((group) => (
          <Card.Body>
            <Card.Title>{group.leader}'s Group</Card.Title>
            <Card.Text>
              <ListGroup.Item variant="dark">Other Members:</ListGroup.Item>
              {group.teammates.map((teammate) => (
                <ListGroup.Item variant="flush">{teammate}</ListGroup.Item>
              ))}
            </Card.Text>
            <LinkContainer to={`/group/${group.id}`}>
              <Button>More Info</Button>
            </LinkContainer>
          </Card.Body>
        ))}
      </div>
    );
  }
}
