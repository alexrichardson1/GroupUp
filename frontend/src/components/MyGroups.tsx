import NavBar from "components/NavBar";
import axios from "axios";
import { config } from "Constants";
import { UserContext } from "components/auth/UserContext";
import { Component } from "react";
import { LinkContainer } from "react-router-bootstrap";
import { Button, Card, ListGroup } from "react-bootstrap";
import { GroupT } from "types/types";

interface Props {}

interface State {
  filteredGroups: GroupT[];
}

export default class MyGroups extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      filteredGroups: [],
    };
  }

  static override contextType = UserContext;

  async getGroups(): Promise<GroupT[]> {
    var result: GroupT[] = [];
    await axios
      .get(`${config.API_URL}/group/`)
      .then((res) => {
        const group = res.data;
        result = group;
      })
      .catch((error) => {
        console.error(error);
      });
    return result;
  }

  override async componentDidMount() {
    this.setState({ filteredGroups: await this.getGroups() }, () =>
      this.filterMyGroups()
    );
  }

  filterMyGroups = () => {
    let value = this.context;
    const user = value.value;
    console.log(this.state.filteredGroups);
    console.log("groups ^");
    this.setState({
      filteredGroups: this.state.filteredGroups.filter(
        (gr) => gr.leader === user || gr.teammates.includes(user.split(" ")[0])
      ),
    });
  };

  override render() {
    return (
      <div>
        <NavBar
          renderBool={[false, false, false, false]}
          create={false}
          savedSearches={true}
        />
        <h3 className="groupsHome">Groups you're already in.</h3>
        {this.state.filteredGroups.map((group) => (
          <Card.Body>
            <Card.Title>{group.leader}'s Group</Card.Title>
            <Card.Text>
              <ListGroup.Item variant="dark">Other Members:</ListGroup.Item>
              {group.teammates.map((teammate) => (
                <ListGroup.Item variant="flush">{teammate}</ListGroup.Item>
              ))}
            </Card.Text>
            <LinkContainer to={`/group/${group.id}`}>
              <Button>More Info</Button>
            </LinkContainer>
          </Card.Body>
        ))}
      </div>
    );
  }
}
