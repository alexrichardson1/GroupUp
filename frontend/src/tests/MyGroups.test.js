import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import MyGroups from "components/MyGroups";

test("renders correctly",async  () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MyGroups />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
