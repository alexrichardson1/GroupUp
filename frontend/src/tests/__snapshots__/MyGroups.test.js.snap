// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navBar navbar navbar-expand-lg navbar-light fixed-top"
  >
    <span
      className="navbar-brand"
      style={
        Object {
          "color": "white",
        }
      }
    >
      <img
        alt="group up logo"
        className="logoImage"
        href="/home"
        onClick={[Function]}
        src="https://i.imgur.com/w92t1xH.png"
      />
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="mr=auto navbar-nav"
        onKeyDown={[Function]}
      >
        <span
          className="navbar-brand"
          style={
            Object {
              "color": "rgb(238, 237, 237)",
            }
          }
        >
           
        </span>
      </div>
      <div
        className="ms-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="ml-auto navHelp navbar-brand"
          href="/help"
          id="collasible-nav-dropdown"
          onClick={[Function]}
          title="Help"
        >
          <img
            alt="help"
            className="splitterImage"
            src="/info-circle.svg"
          />
        </a>
      </div>
    </div>
  </nav>
  <h3
    className="groupsHome"
  >
    Groups you're already in.
  </h3>
</div>
`;
